app/controllers/sentences_lists_controller.php

```diff
@@ -414,9 +415,17 @@ public function of_user($username, $filter = null)
         );
         $userLists = $this->paginate('SentencesList');
 
-        $this->set('userLists', $userLists);
         $this->set('username', $username);
+        $this->set('userLists', $userLists);
         $this->set('filter', $filter);
+        $this->set("userExists", true);
+        
+        $userId = $this->User->getIdFromUsername($username);
+        if (empty($userId)) {
+            $backLink = $this->referer(array('action'=>'index'), true);
+            $this->set('backLink', $backLink);
+            $this->set("userExists", false);
+        }
```

Comment:
Same comment here. Since `$username` is a parameter, you want to check its validity at the very beginning, before even attempting to use it.